edk2toollib/uefi/edk2/parsers/base_parser.py

```diff
@@ -42,6 +42,8 @@ def __init__(self, log="BaseParser"):
         self.PPs = []
         self._Edk2PathUtil = None
         self.TargetFilePath = None  # the abs path of the target file
+        self.FilePathStack = []  # a stack containing a list of size 2: [filepath, line_count]
```

Comment:
NIT: 

Since you're likely dealing with a large set of data, the following solution probably doesn't make a lot of sense performance wise but if you wanted to improve readability of the code you could do something like a namedtuple.

However the object instantiation might be too much for a large dataset

```python

from collections import namedtuple

# Define the namedtuple type
FilePathRecord = namedtuple('FilePathRecord', ['filepath', 'line_count'])

self.FilePathStack = []  # a stack containing FilePathRecord instances

self.FilePathStack.append(FilePathRecord('my_file_path', 100))

self.FilePathStack[0].file_path
```